### PR TITLE
Prepare for upstream removal of getAlignment

### PR DESCRIPTION
**Pull request by dstutt**
getAlignment -> getAlign (and returning Align type).
Prepare for this upstream change

### CODE REVIEW

File: lgc/patch/PatchLoadScalarizer.cpp [hunk 1]
```diff
@@ -182,7 +182,13 @@ void PatchLoadScalarizer::visitLoadInst(LoadInst &loadInst) {
       Value *loadCompPtr = m_builder->CreateConstGEP1_32(compTy, newLoadPtr, i,
                                                          loadInst.getPointerOperand()->getName() + ".i" + Twine(i));
       // Calculate the alignment of component i
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 426675
+      // Old version of the code
       uint64_t compAlignment = MinAlign(loadInst.getAlignment(), i * compSize);
+#else
+      // New version of the code (also handles unknown version, which we treat as latest)
+      uint64_t compAlignment = MinAlign(loadInst.getAlign().value(), i * compSize);
+#endif
```
Comment (Flakebi) [inline lgc/patch/PatchLoadScalarizer.cpp:191]: Calling `.value()` on an alignment is discouraged, it’s better to keep the values as `Align`. You can do
```cpp
Align compAlignment = commonAlignment(loadInst.getAlign(), Align(i * compSize));
```

instead. Also, I think the if guards are not needed.

Comment (dstutt) [reply lgc/patch/PatchLoadScalarizer.cpp:191]: Hmm, I get an assertion with those changes - obviously something else is required. I'll dig further.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 60cf3d50ecd7f8eb2fc062ab6004e005f20e7418
<details><summary>CTS tests (Failed: 0/186371)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36631/65208 (56.2%)</li>
<li>Failed: 0/65208 (0.0%)</li>
<li>Not Supported: 28577/65208 (43.8%)</li><li>Warnings: 0/65208 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29989/55953 (53.6%)</li>
<li>Failed: 0/55953 (0.0%)</li>
<li>Not Supported: 25964/55953 (46.4%)</li><li>Warnings: 0/55953 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38930/65210 (59.7%)</li>
<li>Failed: 0/65210 (0.0%)</li>
<li>Not Supported: 26280/65210 (40.3%)</li><li>Warnings: 0/65210 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 0bcc4e9d89c0262fc24341c2ead3c7c768441091
<details><summary>CTS tests (Failed: 1/172728)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36631/65208 (56.2%)</li>
<li>Failed: 0/65208 (0.0%)</li>
<li>Not Supported: 28577/65208 (43.8%)</li><li>Warnings: 0/65208 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 25664/42310 (60.7%)</li>
<li><details><summary>Failed: 1/42310 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.robustness.image_robustness.bind.notemplate.rgba32f.unroll.nonvolatile.sampled_image.no_fmt_qual.img.samples_1.1d_array.vert
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 16645/42310 (39.3%)</li><li>Warnings: 0/42310 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38930/65210 (59.7%)</li>
<li>Failed: 0/65210 (0.0%)</li>
<li>Not Supported: 26280/65210 (40.3%)</li><li>Warnings: 0/65210 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6c0f1c7b19506291b616375d87b40254b36af94f
<details><summary>CTS tests (Failed: 0/186371)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36631/65208 (56.2%)</li>
<li>Failed: 0/65208 (0.0%)</li>
<li>Not Supported: 28577/65208 (43.8%)</li><li>Warnings: 0/65208 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29989/55953 (53.6%)</li>
<li>Failed: 0/55953 (0.0%)</li>
<li>Not Supported: 25964/55953 (46.4%)</li><li>Warnings: 0/55953 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38930/65210 (59.7%)</li>
<li>Failed: 0/65210 (0.0%)</li>
<li>Not Supported: 26280/65210 (40.3%)</li><li>Warnings: 0/65210 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**